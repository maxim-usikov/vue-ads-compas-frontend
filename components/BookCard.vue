<template>
  <VCard color="gray" hover>
    <VLayout row>
      <VFlex>
        <VCardTitle primary-title>
          <div>
            <div class="headline">{{ title }}</div>
          </div>
        </VCardTitle>
      </VFlex>
    </VLayout>

    <VDivider light />

    <VCardActions class="pa-3">
      <VChip v-for="author in authors" :key="author.id" class="mr-3">{{
        author.name
      }}</VChip>
    </VCardActions>
  </VCard>
</template>

<script>
export default {
  name: 'BookCard',

  props: {
    title: {
      type: String,
      required: true
    },
    authors: {
      type: Array,
      required: true
    }
  }
}
</script>
