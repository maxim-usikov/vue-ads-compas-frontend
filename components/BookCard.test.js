import { shallowMount } from '@vue/test-utils'

import BookCard from './BookCard.vue'

describe('BookCard', () => {
  let $c

  beforeEach(() => {
    $c = shallowMount(BookCard, {
      propsData: { title: 'test book', authors: [] }
    })
  })

  it('is a Vue instance', () => {
    expect($c.isVueInstance()).toBeTruthy()
  })

  it('name is `BookCard`', () => {
    expect($c.name()).toBe('BookCard')
  })

  it('renders correctly', () => {
    expect($c.element).toMatchSnapshot()
  })

  it('renders correctly with authors not empty', () => {
    $c.setProps({ authors: [{ id: 111, name: 'test author' }] })

    expect($c.element).toMatchSnapshot()
  })
})
