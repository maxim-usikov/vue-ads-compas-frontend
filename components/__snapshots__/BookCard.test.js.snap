// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookCard renders correctly 1`] = `
<vcard-stub
  color="gray"
  hover="true"
  tag="div"
>
  <vlayout-stub
    row=""
    tag="div"
  >
    <vflex-stub
      tag="div"
    >
      <vcardtitle-stub
        primarytitle="true"
      >
        <div>
          <div
            class="headline"
          >
            test book
          </div>
        </div>
      </vcardtitle-stub>
    </vflex-stub>
  </vlayout-stub>
   
  <vdivider-stub
    light="true"
  />
   
  <vcardactions-stub
    class="pa-3"
  />
</vcard-stub>
`;

exports[`BookCard renders correctly with authors not empty 1`] = `
<vcard-stub
  color="gray"
  hover="true"
  tag="div"
>
  <vlayout-stub
    row=""
    tag="div"
  >
    <vflex-stub
      tag="div"
    >
      <vcardtitle-stub
        primarytitle="true"
      >
        <div>
          <div
            class="headline"
          >
            test book
          </div>
        </div>
      </vcardtitle-stub>
    </vflex-stub>
  </vlayout-stub>
   
  <vdivider-stub
    light="true"
  />
   
  <vcardactions-stub
    class="pa-3"
  >
    <vchip-stub
      class="mr-3"
      value="true"
    >
      test author
    </vchip-stub>
  </vcardactions-stub>
</vcard-stub>
`;
